cmd/data-scanner.go

```diff
@@ -250,7 +279,7 @@ var globalScannerStats scannerStats
 // The returned cache will always be valid, but may not be updated from the existing.
 // Before each operation sleepDuration is called which can be used to temporarily halt the scanner.
 // If the supplied context is canceled the function will return at the first chance.
-func scanDataFolder(ctx context.Context, poolIdx, setIdx int, basePath string, cache dataUsageCache, getSize getSizeFn) (dataUsageCache, error) {
+func scanDataFolder(ctx context.Context, poolIdx, setIdx int, basePath string, cache dataUsageCache, getSize getSizeFn, scanMode madmin.HealScanMode) (dataUsageCache, error) {
```

Comment:
Instead of splatting this parameter everywhere send it in the cache.

Reply:
> Instead of splatting this parameter everywhere send it in the cache.

I had this thought but then it is weird to add it in 'dataUsageCache'. I will do anyway.